src/solvers/smt2/smt2_conv.cpp

```diff
@@ -2316,6 +2316,20 @@ void smt2_convt::convert_typecast(const typecast_exprt &expr)
       const typecast_exprt tmp(src, bool_typet());
       convert_typecast(typecast_exprt(tmp, dest_type));
     }
+    else if(src_type.id() == ID_bv)
+    {
+      if(
+        to_bv_type(src_type).get_width() !=
+        to_floatbv_type(dest_type).get_width())
+      {
+        UNEXPECTEDCASE("Typecast bv -> float with wrong width");
```

Comment:
I think this is a `DATA_INVARIANT` of  typecast.  `UNHANDLED_CASE` was intended for `case` statements and as a route for converting old code that `case OTHER : assert(0);`.

Reply:
@kroening I believe this is still open.

Reply:
I don't think we guarantee that -- we do commonly cast between representations of different widths, say a cast from unsignedbv[8] to signedbv[32] is extremely common.

Reply:
Isn't that represented as a BV typecast though?

Reply:
We should be able to handle this by simply doing a BV typecast, then a BV to FP one.

Reply:
For this particular case, given that "bv" has no interpretation, it would probably be sensible to require that the width must match.
However, we don't have any of these at the moment, so this would be dead code either way.

Reply:
Fair.